addons/interaction/XEH_postInit.sqf

```diff
@@ -135,3 +135,20 @@ GVAR(isOpeningDoor) = false;
         }];
     };
 }] call CBA_fnc_addEventHandler;
+
+
+// add "Take _weapon_" action to dropped weapons
+private _action = [
+    // action display name will be overwritten in modifier function
+    QGVAR(takeWeapon), "take", "\A3\ui_f\data\igui\cfg\actions\take_ca.paa",
+    {_player action ["TakeWeapon", _target, weaponCargo _target select 0]},
+    {1 == count weaponCargo _target},
+    nil, nil, nil, nil, nil,
+    {
+        params ["_target", "", "", "_actionData"];
+        _actionData set [1, format [localize "STR_ACTION_TAKE_BAG", getText (configfile >> "CfgWeapons" >> weaponCargo _target select 0 >> "displayName")]];
+    }
+] call EFUNC(interact_menu,createAction);
+{
+    [_x, 0, ["ACE_MainActions"], _action, true] call EFUNC(interact_menu,addActionToClass);
+} forEach ["WeaponHolder", "WeaponHolderSimulated"];
```

Comment:
Use https://github.com/CBATeam/CBA_A3/pull/1228?

Reply:
Looks like I wanted not to create extra function for action modifying. So no changes are required here. Anyway I removed XEH creating lines.